scripts/dockerfiles/polkadot/polkadot_builder.Dockerfile

```diff
@@ -1,27 +1,35 @@
+# This is the build stage for Polkadot. Here we create the binary in a temporary image.
 FROM docker.io/paritytech/ci-linux:production as builder
-LABEL io.parity.image.description="This is the build stage for Polkadot. Here we create the binary."
 
 WORKDIR /polkadot
-
 COPY . /polkadot
 
-RUN cargo build --release --locked
-
-# ===== SECOND STAGE ======
+RUN cargo build --locked --release
 
+# This is the 2nd stage: a very small image where we copy the Polkadot binary."
 FROM docker.io/library/ubuntu:20.04
-LABEL io.parity.image.description="Polkadot: a platform for web3. This is a self-buit multistage image."
+
+LABEL description="Multistage Docker image for Polkadot: a platform for web3" \
+	io.parity.image.type="builder" \
+	io.parity.image.authors="chevdor@gmail.com, devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.description="Polkadot: a platform for web3" \
+	io.parity.image.source="https://github.com/paritytech/polkadot/blob/${VCS_REF}/scripts/dockerfiles/polkadot/polkadot_builder.Dockerfile" \
+	io.parity.image.documentation="https://github.com/paritytech/polkadot/"
 
 COPY --from=builder /polkadot/target/release/polkadot /usr/local/bin
 
 RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadot polkadot && \
-	mkdir -p /polkadot/.local/share && \
-	mkdir /data && \
+	mkdir -p /data /polkadot/.local/share && \
 	chown -R polkadot:polkadot /data && \
-	ln -s /data /polkadot/.local/share/polkadot && \
-	rm -rf /usr/bin /usr/sbin
+	ln -s /data /polkadot/.local/share/polkadot
 
+# check if executable works in this container
+RUN /usr/local/bin/polkadot --version
+
+RUN	rm -rf /usr/bin /usr/sbin
```

Comment:
why do you want to remove `/usr/bin` from the image that's intended to be used in the local experiments?

Reply:
also padding

Reply:
I had this removal initially and decided to keep it but I did consider removing (.. the rm).

> from the image that's intended to be used in the local experiments?

Who says that? Users may use the image for `subkey` for instance.

Here are the pros/cons or the removal:

## pros
- safer
- (slightly) smaller image
- the image is made to use the binaries we provide, not to fiddle around with bash, curl & Co.
- safe by default but users can comment out the line and build an image where those are not removed

## cons
- cannot bash into the container
- may surprise some users
- admitedly, this container is not running 24/7 and will be short lived so we may not need the extra safety

As conclusion, I think it is better to go safer by default and avoid making assumptions about **how** users will use the image. Still users can easily get more freedom by commenting out the line.

I would be fine removing this `rm` if you can provide a sample use case where we'd need something in `/usr/bin` while using this image.

Reply:
Wait, I thought `builder_*.Dockerfile` are designed for the developers to build a binary in a reproducible env to play with it in the first place. To not have to setup the exact Rust versions and other deps. Not to run as a validator, etc.
I realize that actually we can't tell or control its' usage, it's just the main usecase is dependent on how we advertise it.

Reply:
> are designed for the developers to build a binary in a reproducible env to play with it in the first place.

I definitely don't think this the only use case. I also see the case of validators who want to build their own binary + image themselves.

Reply:
OK, sounds good to me. Then let's just do the following.